File: zombie.ml
let rec get_time total speed c f x =
    let gap = x -. c in
    let gapf = gap *. f and xgap = x -. gap in
    if xgap *. speed >= gapf then
        total +. x /. speed
    else
        get_time (total +. c /. speed) (speed +. f) c f x ;;

let x = get_time 0.0 2. 37. 7. 54500000000. in
print_float x ;;

File: zombie.c
#include <stdio.h>

double get_time(double c, double f, double x) {
	double gap = x - c;
	double gapf = gap * f;
	double xgap = x - gap;

    double total = 0.0;
    double speed = 2.0;

    while(xgap * speed < gapf) {
        total += c / speed;
        speed += f;
    }
    total += x / speed;
    return total;
}

int main(void) {
    printf("%e\n", get_time(37, 7, 54500000000.0));
	return 0;
}
